x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPolicyRunner.java

```diff
@@ -194,11 +195,56 @@ static void validateMappings(
         }
     }
 
-    private static void validateField(Map<?, ?> properties, String fieldName, boolean fieldRequired) {
+    private record MappingTypeAndFormat(String type, String format) {
+
+    }
+
+    private static MappingTypeAndFormat getMappingTypeAndFormat(
+        String fieldName,
+        EnrichPolicy policy,
+        boolean strictlyRequired,
+        List<Map<String, Object>> sourceMappings
+    ) {
+        var fieldMappings = sourceMappings.stream()
+            .map(mapping -> getMappingTypeAndFormat(mapping, fieldName, strictlyRequired))
+            .filter(Objects::nonNull)
+            .toList();
+        Set<String> types = fieldMappings.stream().map(tf -> tf.type).collect(Collectors.toSet());
+        if (types.size() > 1) {
+            if (strictlyRequired) {
+                throw new ElasticsearchException(
+                    "Multiple distinct mapping types for field '{}' - indices({})  types({})",
+                    fieldName,
+                    Strings.collectionToCommaDelimitedString(policy.getIndices()),
+                    Strings.collectionToCommaDelimitedString(types)
+                );
+            }
+            return null;
+        }
+        if (types.isEmpty()) {
+            return null;
+        }
+        Set<String> formats = fieldMappings.stream().map(tf -> tf.format).filter(Objects::nonNull).collect(Collectors.toSet());
+        if (formats.size() > 1) {
+            if (strictlyRequired) {
+                throw new ElasticsearchException(
+                    "Multiple distinct formats specified for field '{}' - indices({})  format entries({})",
+                    policy.getMatchField(),
+                    Strings.collectionToCommaDelimitedString(policy.getIndices()),
+                    Strings.collectionToCommaDelimitedString(formats)
+                );
+            }
+            return null;
+        }
+        return new MappingTypeAndFormat(Iterables.get(types, 0), formats.isEmpty() ? null : Iterables.get(formats, 0));
+    }
+
+    @SuppressWarnings("unchecked")
```

Comment:
is it possible to not not suppress unchecked warnings?

Reply:
++, I pushed https://github.com/elastic/elasticsearch/pull/96056/commits/db42ed0dc2c37006ca9424920790845edb30a9d7 to reduce the scope of the suppression.